luigi/static/visualiser/js/visualiserApp.js

```diff
@@ -50,35 +80,145 @@ function visualiserApp(luigi) {
+            infoBox.dataset.on = 'yes';
+            infoBoxIcon.removeClass(colorClass);
+            $(infoBox).addClass(colorClass);
+        }
+        else {
+            infoBox.dataset.on = 'no';
+            $(infoBox).removeClass(colorClass);
+            infoBoxIcon.addClass(colorClass);
+        }
+    }
+
+    function renderSidebar(tasks) {
+        // tasks is a list of task names
+        var counts = {};
+        $.each(tasks, function(i) {
+            var name = tasks[i];
+            if (counts[name] === undefined) {
+                counts[name] = 0;
             }
-            indexedTasks[task[fieldName]].push(task);
+            counts[name] += 1;
+        });
+        var taskList = [];
+        $.each(counts, function (name) {
+            taskList.push({name: name, count: counts[name]});
         });
-        return indexedTasks;
+        return renderTemplate("sidebarTemplate", {"tasks": taskList});
     }
 
-    function entryList(object) {
-        var list = [];
-        $.each(object, function(key, value) {
-            list.push({key: key, value:value});
+    function selectSidebarItem(item) {
+        var sidebarItems = $('.sidebar').find('li');
+        sidebarItems.each(function (i) {
+            var item2 = sidebarItems[i];
+            if (item2.dataset.task === undefined) {
+                return;
+            }
+            if (item === item2) {
+                if ($(item2).hasClass('active')) {
+                    // item is active, deselect
+                    $(item2).removeClass('active');
+                    $(item2).find('.badge').removeClass('bg-green');
+                }
+                else {
+                    // select item
+                    $(item2).addClass('active');
+                    $(item2).find('.badge').addClass('bg-green');
+                }
+            }
+            else {
+                // clear any selection
+                $(item2).removeClass('active');
+                $(item2).find('.badge').removeClass('bg-green');
+            }
         });
-        return list;
     }
 
-    function renderTasks(tasks) {
-        var displayTasks = tasks.map($.proxy(taskToDisplayTask, null, true));
-        displayTasks.sort(function(a,b) { return b.displayTimestamp - a.displayTimestamp; });
-        var tasksByFamily = entryList(indexByProperty(displayTasks, "taskName"));
-        tasksByFamily.sort(function(a,b) { return a.key.localeCompare(b.key); });
-        return renderTemplate("rowTemplate", {tasks: tasksByFamily});
+    function renderWarnings() {
+        return renderTemplate("warningsTemplate",
+            {missingCategories: $.map(missingCategories, function (v, k) {return v})}
+        );
     }
 
     function processWorker(worker) {
-        worker.tasks = worker.running.map($.proxy(taskToDisplayTask, null, false));
+        worker.tasks = worker.running.map($.proxy(taskToDisplayTask, false, null));
```

Comment:
This call does not work because $.proxy is weird. Luckily, we don't need the proxy function now that you've changed the order of the arguments to taskToDisplayTask and made the second one optional. Just do

    worker.tasks = worker.running.map(taskToDisplayTask);

and that'll fix it.